src/core/border_router/routing_manager.cpp

```diff
@@ -2166,6 +2166,39 @@ RoutingManager::OnLinkPrefixManager::OnLinkPrefixManager(Instance &aInstance)
     mOldLocalPrefixes.Clear();
 }
 
+void RoutingManager::OnLinkPrefixManager::SetState(State aState)
+{
+    VerifyOrExit(mState != aState);
+    mState = aState;
+
+#if OT_SHOULD_LOG_AT(OT_LOG_LEVEL_INFO)
+    {
+        const char *actionText = "";
+
+        switch (mState)
```

Comment:
We seem to have established a `StateToString` design pattern. Should we follow that here too?

Reply:
Good suggestion. Changed in new push.